packages/backend-for-frontend/src/services/clientService.ts

```diff
@@ -422,20 +418,24 @@ export async function decorateKey(
   members: string[],
   correlationId: CorrelationId
 ): Promise<bffApi.PublicKey> {
-  const user = await getSelfcareUserById(
-    selfcareClient,
-    key.userId,
-    selfcareId,
-    correlationId
-  );
-
-  return {
-    user: toBffApiCompactUser(user, key.userId),
-    name: key.name,
-    keyId: key.kid,
-    createdAt: key.createdAt,
-    isOrphan: !members.includes(key.userId) || user.id === undefined,
-  };
+  try {
+    const user = await getSelfcareUserById(
+      selfcareClient,
+      key.userId,
+      selfcareId,
+      correlationId
+    );
+
+    return {
+      user: toBffApiCompactUser(user, key.userId),
+      name: key.name,
+      keyId: key.kid,
+      createdAt: key.createdAt,
+      isOrphan: !members.includes(key.userId) || user.id === undefined,
+    };
+  } catch (error) {
+    throw userNotFound(key.userId, selfcareId);
+  }
```

Comment:
```suggestion
    const user = await getSelfcareUserById(
      selfcareClient,
      key.userId,
      selfcareId,
      correlationId
    );

    return {
      user: toBffApiCompactUser(user, key.userId),
      name: key.name,
      keyId: key.kid,
      createdAt: key.createdAt,
      isOrphan: !members.includes(key.userId) || user.id === undefined,
    };
```
this try/catch should be removed too